### ISSUE TITLE
Cannot reference built-in abstract base classes without type parameters

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/pyright#6582]
[Playground Link](https://pyright-play.net/?strict=true&enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMpoCmMA%2BrkmqgFBY55QDGYANs4QzEmCgM4B0AQwBGDKlQCyATRIBlAKIAVGQC4oPYgG0wQgFbsYAXSgBeKAG8qUK2uIAaS9aat9XXoJF8ZdqgF8xCEFQYAAoiUnJKFGD1GA0ggwBKTBQoKVlFGSSAYigFEABXQioAoNDiMkDI4Kc2Dld%2BYQZPTXik1FTpeSVs3ILCIA)

Using `"typeCheckingMode": "strict"`:

```python
from typing import get_origin
import collections.abc

MY_SETS: set[object] = {
    set,
    collections.abc.Set,
#   ~~~~~~~~~~~~~~~~~~~
# Type of "Set" is partially unknown
#  Type of "Set" is "type[AbstractSet[Unknown]]"
}

print(get_origin(set[int]) in MY_SETS) # True
print(get_origin(collections.abc.Set[int]) in MY_SETS) # True
```

Currently the sample above results in the following error:

```
Type of "Set" is partially unknown
  Type of "Set" is "type[AbstractSet[Unknown]]"
```

Here I was actually trying to leverage the runtime value of `collections.abc.Set` - I don't *believe* I need type arguments for this to work correctly.

A reasonable workaround would be to just use `get_origin(collections.abc.Set[object])`, but I figured it'd be worth raising.

**Comment by erictraut** [comment]
Thanks for the bug report. I agree there's an inconsistency here. If you use `from collections.abc import Set` and use `Set` in place of `collections.abc.Set`, the error is not reported. These are equivalent, so they should be consistent.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
